tests/Middlewares/Laravel/VerifySignatureTest.php

```diff
@@ -132,6 +138,82 @@ public function it_should_call_our_callback_if_the_request_is_valid()
         });
     }
 
+    /**
+     * @test
+     */
+    public function it_should_set_the_key_when_one_is_passed()
+    {
+        $id = (string) Uuid::uuid4();
+
+        $this->configurations->shouldReceive('get')
+            ->with('signed-requests.custom.headers.signature')
+            ->andReturn('signature');
+
+        $this->configurations->shouldReceive('get')
+            ->with('signed-requests.custom.headers.algorithm')
+            ->andReturn('algorithm');
+
+        $this->configurations->shouldReceive('get')
+            ->with('signed-requests.custom.key')
+            ->andReturn('key');
+
+        $this->configurations->shouldReceive('get')
+            ->with('signed-requests.custom.request-replay.allow')
+            ->andReturn(true);
+
+        $this->configurations->shouldReceive('get')
+            ->with('signed-requests.custom.cache-prefix')
+            ->andReturn('prefix');
+        $this->configurations->shouldReceive('get')
+            ->with('signed-requests.custom.request-replay.tolerance')
+            ->andReturn(60);
+
+        $query = [];
+        $request = [];
+        $attributes = [];
+        $cookies = [];
+        $files = [];
+        $server = [
+            'HTTP_X-SIGNED-ID' => $id,
+            'HTTP_ALGORITHM' => 'sha256'
+        ];
+
+        $request = new Request($query, $request, $attributes, $cookies, $files, $server, 'a');
+        $request->headers->set('signature', (string) new Signature(new Payload($request), 'sha256', 'key'));
+
+        $this->middleware->handle($request, function () {
+            // This should be called.
+            $this->assertTrue(true);
```

Comment:
There is no guarantee that this is called. If we did something like
```PHP
$called = false;
$this->middleware->handle($request, function () use (&$called) {
    $called = true;
}, 'custom');
$this->assertTrue($called);
```

That way if the closure is never executed, `called` will be false and the assertion will fail. Currently if the closure is not executed, it will just not perform that assertion, which won't necessarily cause the test to fail.